js/index.js

```diff
@@ -1,35 +1,39 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  const price = product.querySelector(".price span").innerText;
+  const quantity = product.querySelector(".quantity input").value;
+  const subTotalPrice = quantity * price;
+  const subTotal = product.querySelector(".subtotal span");
+  subTotal.innerText = subTotalPrice;
+  return subTotalPrice;
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
-  // ITERATION 2
-  //... your code goes here
+// ITERATION 2-3
 
-  // ITERATION 3
-  //... your code goes here
+function calculateAll() {
+  let priceTotal = 0;
+  const priceTotalInnerText = document.querySelector("#total-value span");
+  const productArray = [...document.getElementsByClassName("product")];
+  for (let i=0; i < productArray.length; i++){
+      updateSubtotal(productArray[i]);
+      priceTotal += updateSubtotal(productArray[i]);
+  }
+  priceTotalInnerText.innerText = priceTotal;
 }
 
 // ITERATION 4
+const removeButtons = [...document.getElementsByClassName("btn btn-remove")];
+removeButtons.forEach(element => {element.addEventListener("click", removeProduct)});
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  target.parentNode.parentNode.querySelector(".quantity input").value = 0;
```

Comment:
I think you should remove the target.parentNode.parentNode - not set it's quantity input value to 0.

Reply:
so
```js
event.currentTarget.parentNode.parentNode.remove()
```

Reply:
Yeah, I know. I misunderstood the task here and only realized that on Thursday during our class when we were recapping  this iteration